homeassistant/components/netatmo/data_handler.py

```diff
@@ -0,0 +1,185 @@
+            self._data_classes[CAMERA_DATA_CLASS_NAME][NEXT_SCAN] = time()
+
+    async def async_fetch_data(self, data_class, data_class_entry, **kwargs):
+        """Fetch data and notify."""
+        try:
+            self.data[data_class_entry] = await self.hass.async_add_executor_job(
+                partial(data_class, **kwargs), self._auth,
+            )
+            for update_callback in self._data_classes[data_class_entry][
+                "subscriptions"
+            ]:
+                if update_callback:
+                    update_callback()
+        except (pyatmo.NoDevice, pyatmo.ApiError) as err:
+            _LOGGER.debug(err)
+
+    async def register_data_class(
+        self, data_class_name, data_class_entry, update_callback, **kwargs
+    ):
+        """Register data class."""
+        async with self.lock:
+            if data_class_entry not in self._data_classes:
+                self._data_classes[data_class_entry] = {
+                    "class": DATA_CLASSES[data_class_name],
+                    "name": data_class_entry,
+                    "interval": DEFAULT_INTERVALS[data_class_name],
+                    NEXT_SCAN: time() + DEFAULT_INTERVALS[data_class_name],
+                    "kwargs": kwargs,
+                    "subscriptions": [update_callback],
+                }
+
+                await self.async_fetch_data(
+                    DATA_CLASSES[data_class_name], data_class_entry, **kwargs
+                )
+
+                self._queue.append(self._data_classes[data_class_entry])
+                _LOGGER.debug("Data class %s added", data_class_entry)
+
+            else:
+                self._data_classes[data_class_entry]["subscriptions"].append(
+                    update_callback
+                )
+
+    async def unregister_data_class(self, data_class_entry, update_callback):
+        """Unregister data class."""
+        # Give the entities a chance to subscribe before removing the data class entry
+        # so that the data can be reused and reduce the number of API calls
+        await asyncio.sleep(1)
+
+        async with self.lock:
+            self._data_classes[data_class_entry]["subscriptions"].remove(
+                update_callback
+            )
+
+            if not self._data_classes[data_class_entry].get("subscriptions"):
+                self._queue.remove(self._data_classes[data_class_entry])
+                self._data_classes.pop(data_class_entry)
+                _LOGGER.debug("Data class %s removed", data_class_entry)
+
+    @property
+    def webhook(self) -> bool:
+        """Return the webhook state."""
+        return self._webhook
+
+
+@callback
+def add_entities(entities, async_add_entities, hass):
+    """Add new sensor entities."""
+    async_add_entities(entities)
+
+
+async def async_config_entry_updated(hass: HomeAssistant, entry: ConfigEntry) -> None:
+    """Handle signals of config entry being updated."""
+    async_dispatcher_send(hass, "signal_update")
```

Comment:
I'm pretty sure the signal name needs to be prefixed with `neatmo` as I recall these being globals.

Need to double check this